ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -1307,6 +1346,59 @@ private class JdbcLedgerDao(
     )
   }
 
+  override def getPackageUploadEntries(
+      startInclusive: Long,
+      endExclusive: Long): Source[(Long, PackageUploadEntry), NotUsed] =
+    paginatingStream(
+      startInclusive,
+      endExclusive,
+      PageSize,
+      (startI, endE) => {
+        dbDispatcher.executeSql(s"load package upload entries", Some(s"bounds: [$startI, $endE[")) {
+          implicit conn =>
+            SQL_SELECT_PACKAGE_UPLOAD_ENTRIES
+              .on("startInclusive" -> startI, "endExclusive" -> endE)
+              .as(packageUploadEntryParser.*)
+        }
+      }
+    ).flatMapConcat(Source(_))
+
+  override def storePackageUploadEntry(
+      offset: LedgerOffset,
+      newLedgerEnd: LedgerOffset,
+      externalOffset: Option[ExternalOffset],
+      participantId: ParticipantId,
+      submissionId: String,
+      reason: Option[String],
+      typ: String): Future[PersistenceResponse] = {
```

Comment:
Please make the accept/reject case typed and use e.g. `case object`s or an `Enumeration` instead of `typ: String`.

Reply:
The `typ` can also be derived from whether a `reason` is `Some` or `None`, no?

Reply:
added use of PackageUploadEntry type